### PR TITLE
Added FluidMixingEvent

### PR DESCRIPTION
**Pull request by WaDelma**
This is a Event that I created for #1090.
I tried to fix the problems that I presented in the issue.

New pull request because I derped the last one...

I am currently creating FluidMixingEvent for Fluid API. This would be send whenever two different fluids are tried to combine.

With this event you could create custom interactions between liquids in machinery. My motivation for this event is to enable of usage of NBT data in FluidStack for functionality instead of it just describing a new liquid.

The problems that I am facing currently:
* I am extending FluidEvent which expects one FluidStack in its constructor, but FluidMixingEvent would need to have to 2 of them. One for source and one for target... Except filling/draining fluid events actually don't need their FluidStack as you can get it from the ITank. This is why I am not sure what to give in the super call. Also should I edit those other FluidEvents?
* First I thought that equality of fluids for denying different fluids mixing was only called in FluidTanks fill method, but I noticed that also TileFluidHandlers drain method with FluidStack parameter does it too. Should I be changing both of those and how should I let people know it in Javadoc?

### CODE REVIEW

**Comment by WaDelma** [comment]
I am actually shocked that rebasing this only had one conflict and it was in imports.

**Comment by cpw** [comment]
Nice @WaDelma, thanks for updating. Sadly it's targetting the wrong branch, but that's not necessarily a problem.

**Comment by cpw** [comment]
We've taken a new look at this, and I think that the concept of having tanks always mixing, is probably extremely unexpected. It would be much better accomplished through a custom tank implementation, as such I think that this event is unnecessary since nothing in forge would ever use it.

**Comment by WaDelma** [comment]
Well the problem is interaction with other mods... Basically that would mean the liquid could only be transported with that mods system.

**Comment by LexManos** [comment]
The issue I think is you need to articulate exactly what you want more.
What I THINK you want, opposed to what this PR actually is.
Is a way to more carefully determine if two fluid stacks are equal when inserting into a tank.
This has nothing to do with custom mixing this has everything to do with NOT mixing when you don't want to.
At least what I think you're getting at.

If that IS the case this is definitely NOT the way to go.
Please articulate what you want better.

**Comment by WaDelma** [comment]
Okay, I'll try.

Basically this pr started from the modjam mod I mentioned.
I wanted to create a color liquid that contains RGB value to indicate the color and I wanted to pipe it around.
So I created custom liquid and added RGB value as a tag.
I created pipe that functioned as any other pipe in other mods.
Now when trying to pipe it around I noticed that it wouldn't work if I had multiple different colored liquids simultaneously going through the pipes.
This is because currently the liquid API reqards same liquid with different tags different liquids.
Now when thinking how to make it work, I realised that there would have to be a way determine the color of liquid after one color mixes with another color.
Basically only allowing same liquid with different tags to be reqarded as same doesn't cut it, because there would be no way to preserve the information in tags in meaningful way. What happens when red liquid mixes with blue? If you just make those two liquids equal the result would be blue. If you can make your own custom logic, you can make the result to be purple.

**Comment by WaDelma** [comment]
Does that explanation make any sense? @LexManos
